src/Data/Array.purs

```diff
@@ -781,6 +782,31 @@ fold = F.fold
 intercalate :: forall a. Monoid a => a -> Array a -> a
 intercalate = F.intercalate
 
+-- | The 'transpose' function transposes the rows and columns of its argument.
+-- | For example,
+-- |
+-- | ```purescript
+-- | transpose [[1,2,3], [4,5,6]] == [[1,4], [2,5], [3,6]]
+-- | ```
+-- |
+-- | If some of the rows are shorter than the following rows, their elements are skipped:
+-- |
+-- | ```purescript
+-- | transpose [[10,11], [20], [30,31,32]] == [[10,20,30], [11,31], [32]]
```

Comment:
Could examples docs be changed, so that the underlying matrix is better visualized? For example
```
transpose
  [ [ 1, 2, 3 ]
  , [ 4, 5, 6 ]
  ] ==
  [ [ 1, 4 ]
  , [ 2, 5 ]
  , [ 3, 6 ]
  ]

Reply:
Or even
```purs
(transpose
  [ [ 1, 2, 3 ]
  , [ 4, 5, 6 ]
  ]) == (
  [ [ 1, 4 ]
  , [ 2, 5 ]
  , [ 3, 6 ]
  ])

Reply:
OK - good idea.